src/universal/mutations/RemoveReflectionMutation.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * Removes a reflection for the retrospective meeting.
+ *
+ * @flow
+ */
+import {commitMutation} from 'react-relay';
+import {Environment, RecordProxy, RecordSourceSelectorProxy} from 'relay-runtime';
+
+type Variables = {
+  reflectionId: string,
+};
+
+type CompletedHandler = (response: ?Object, errors: ?Array<Error>) => void;
+
+type ErrorHandler = (error: Error) => void;
+
+graphql`
+  fragment RemoveReflectionMutation_team on RemoveReflectionPayload {
+    meeting {
+      id
+    }
+    reflection {
+      id
+    }
+  }
+`;
+
+const mutation = graphql`
+  mutation RemoveReflectionMutation($reflectionId: ID!) {
+    removeReflection(reflectionId: $reflectionId) {
+      ...RemoveReflectionMutation_team @relay(mask: false)
+    }
+  }
+`;
+
+export const removeReflectionUpdater = (payload: RecordProxy, store: RecordSourceSelectorProxy) => {
+  // TODO - maeby https://github.com/dan-f/maeby
+  if (!payload) {
```

Comment:
same comments as the create mutation: refactor the error & complete handlers so we can reuse them, reuse the patterns in other handlers to make this a little cleaner, replace the optimisticResponse with an updater that contains an object similar to what the server will return (this will prevent really annoying errors that only occur in high-latency conditions)

Reply:
Yeah, I agree this could use a bit of a refactor. I actually think that most of the noisy code is just the falsy testing, since attempting to access any document in the store could return nothing. I was thinking that a monadic construction using something like `maeby` could cut down on that noise.